Demo/Demo/SwiftUIComponents/CardVaultViews/UpdateSetupTokenView.swift

```diff
@@ -0,0 +1,76 @@
+import SwiftUI
+
+struct UpdateSetupTokenView: View {
+
+    let setupToken: String
+
+    @State private var cardNumberText: String = "4111 1111 1111 1111"
+    @State private var expirationDateText: String = "01 / 25"
+    @State private var cvvText: String = "123"
+
+    @ObservedObject var cardVaultViewModel: CardVaultViewModel
+    @ObservedObject var baseViewModel: BaseViewModel
+
+    public init(baseViewModel: BaseViewModel, cardVaultViewModel: CardVaultViewModel, setupToken: String) {
+        self.cardVaultViewModel = cardVaultViewModel
+        self.baseViewModel = baseViewModel
+        self.setupToken = setupToken
+    }
+
+    var body: some View {
+        VStack(spacing: 16) {
+            HStack {
+                Text("Vault Card")
+                    .font(.system(size: 20))
+                Spacer()
+            }
+
+            CardFormView(
+                cardNumberText: $cardNumberText,
+                expirationDateText: $expirationDateText,
+                cvvText: $cvvText
+            )
+
+            let card = baseViewModel.createCard(
+                cardNumber: cardNumberText,
+                expirationDate: expirationDateText,
+                cvv: cvvText
+            )
+
+            ZStack {
+                if case .loading = cardVaultViewModel.state.updateSetupTokenResponse {
+                    ProgressView()
+                        .progressViewStyle(CircularProgressViewStyle(tint: Color.white))
+                        .background(Color.black.opacity(0.4))
+                        .cornerRadius(10)
+                        .frame(maxWidth: .infinity)
```

Comment:
Since we use this across several views, can we extract this into a custom view that way we can ensure our ProgressViews are always consistent?

Reply:
addressed in commit 121c4eadb22995e34dbf68e2aee497baa92e2249